audobject/core/object.py

```diff
@@ -453,8 +472,11 @@ def _resolve_value(
             self,
             name: str,
             value: typing.Any,
+            stream: typing.Optional[typing.IO],
     ) -> DefaultValueType:
         if name in self.resolvers:
+            # let resolver know if we write to a stream
+            self.resolvers[name].__dict__[define.STREAM_ATTRIBUTE] = stream
```

Comment:
What is a stream here? What is the other option we normally write to?

Reply:
It's the file stream, if we don't write to a file it is `None`.

Reply:
Would you say the code is easier to understand if we don't pass around the IO stream, but the root folder from which we read / write?

Reply:
Yes, as we have to store the resource file realtive to the YAML root folder anyway, I think it will be easier to understand if we introduce the concept of that root folder.

Reply:
Ok, I changed it accordingly.